draft-irtf-cfrg-frost.md

```diff
@@ -214,22 +214,45 @@ FROST requires the use of a cryptographically secure hash function, generically
 written as H, which functions effectively as a random oracle. For concrete
 recommendations on hash functions which SHOULD be used in practice, see
 {{ciphersuites}}. Using H, we introduce three separate domain-separated hashes,
-H1, H2, and H3, where H1 and H2 map arbitrary byte strings to Scalar elements of
+H1, H2, H3, and H4, where H1, H2, and H4 map arbitrary byte strings to Scalar elements of
 the prime-order group scalar field, and H3 is an alias for H with a domain separator. 
-The details of H1, H2, and H3 vary based on ciphersuite. See {{ciphersuites}}
+The details of H1, H2, H3, and H4 vary based on ciphersuite. See {{ciphersuites}}
 for more details about each.
 
 # Helper functions {#helpers}
 
 Beyond the core dependencies, the protocol in this document depends on the
 following helper operations:
 
+- Nonce generation, {{dep-nonces}}
 - Schnorr signatures, {{dep-schnorr}};
 - Polynomial operations, {{dep-polynomial}};
 - Encoding operations, {{dep-encoding}};
 - Signature binding {{dep-binding-factor}}, group commitment {{dep-group-commit}}, and challenge computation {{dep-sig-challenge}}
 
-This sections describes these operations in more detail.
+These sections describes these operations in more detail.
+
+## Nonce generation {#dep-nonces}
+
+To hedge against a bad RNG, we generate nocnes by sourcing fresh randomness and 
```

Comment:
I would be more specific here; we don't protect against a RNG returning all zero for example, but this does help for RNGs that return predictable randomness.